cli/file_fetcher.rs

```diff
@@ -161,9 +161,15 @@ fn get_validated_scheme(
   }
 }
 
+#[derive(Debug, Copy, Clone)]
+pub enum FetchPermissionsOptionRef<'a> {
```

Comment:
Shouldn't this one be in `cli/cache/mod.rs`? It's a bit concerning that these are spread across various modules. Maybe add `cli/permissions.rs` to colocate all this code?

Reply:
I don't think so because this is the lowest level this is used at.

Reply:
Ah, I see what you mean because there is `FetchPermissionsOption` and the `Ref` version. I'll just put them both here.